### DISCUSSION TITLE
How do I add support for a courier that is not currently supported?

### DISCUSSION THREAD

**Discussion by ezfrag2021** [moralmunky/Home-Assistant-Mail-And-Packages#961] [General]
Is there a template or instructions that will allow me to add additional couriers to the integration? I live in Portugal which is served by Amazon Spain but the emails are sent in Portuguese so the integration doesn't trigger on them. Can I add the Portuguese language triggers somewhere?

Same question with other couriers in my country that are not currently in the list. Can I add them somehow?
Thanks!

**Comment by firstof9** [comment]
All the email searches that are completed are defined in the `const.py` file.

**Reply by ezfrag2021** [reply]
Ok so I can just edit this to add the functionality that I need?

But these aren't collated in some way to expand the base capability of the integration with additional couriers from different countries?

**Reply by firstof9** [reply]
Usually people will submit a PR to add the couriers for their country.

**Reply by ezfrag2021** [reply]
Ah ok. That makes sense. So once I have it working with my configuration I can submit the additions I made to const.py via a PR and if it's accepted it becomes part of the integration. Thanks for the info.
